src/kitsy-script-toolkit.js

```diff
@@ -3,271 +3,153 @@
-    codeFragment (string): Javascript code to inject. You can provide as many
-      code fragment parameters as you want, they'll all be strung together
-      before they're injected.
-
-    Use this function to perform surgery on the Bitsy code. It's a search-and-
-    insert. Use sparingly. Best used to install new dialog functions, new
-    operators, or expose internal variables globally.
+  https://github.com/seleb/bitsy-hacks/wiki/Coding-with-kitsy
 */
 import bitsy from "bitsy";
+import {
+	unique,
+	flatten,
+	inject as utilsInject
+} from "./utils.js";
+
+
+// Examples: inject('names.sprite.set( name, id );', 'console.dir(names)');
+//           inject('names.sprite.set( name, id );', 'console.dir(names);', 'console.dir(sprite);');
+//           inject('names.sprite.set( name, id );', ['console.dir(names)', 'console.dir(sprite);']);
+export function inject(searchString, codeFragments) {
+	var kitsy = kitsyInit();
+	var args = [].slice.call(arguments);
+	codeFragments = flatten(args.slice(1));
+
+	kitsy.queuedInjectScripts.push({
+		searchString: searchString,
+		codeFragments: codeFragments
+	});
+}
+
+// Ex: before('load_game', function run() { alert('Loading!'); });
+//     before('show_text', function run(text) { return text.toUpperCase(); });
+//     before('show_text', function run(text, done) { done(text.toUpperCase()); });
+export function before(targetFuncName, beforeFn) {
+	var kitsy = kitsyInit();
+	kitsy.queuedBeforeScripts[targetFuncName] = kitsy.queuedBeforeScripts[targetFuncName] || [];
+	kitsy.queuedBeforeScripts[targetFuncName].push(beforeFn);
+}
+
+// Ex: after('load_game', function run() { alert('Loaded!'); });
+export function after(targetFuncName, afterFn) {
+	var kitsy = kitsyInit();
+	kitsy.queuedAfterScripts[targetFuncName] = kitsy.queuedAfterScripts[targetFuncName] || [];
+	kitsy.queuedAfterScripts[targetFuncName].push(afterFn);
+}
+
+function kitsyInit() {
+	// return already-initialized kitsy
+	if (bitsy.kitsy) {
+		return bitsy.kitsy;
+	}
 
-export function kitsyInit() {
-	var globals = bitsy;
-	var firstInit = !globals.kitsy; // check if kitsy has already been inited
-
-	// Allow multiple copies of this script to work in one HTML file.
-	globals.queuedInjectScripts = globals.queuedInjectScripts || [];
-	globals.queuedBeforeScripts = globals.queuedBeforeScripts || {};
-	globals.queuedAfterScripts = globals.queuedAfterScripts || [];
-	globals.superFuncs = globals.superFuncs || {};
-	globals.injectsDone = globals.injectsDone || false;
-
-	// Local aliases
-	var queuedInjectScripts = globals.queuedInjectScripts;
-	var queuedBeforeScripts = globals.queuedBeforeScripts;
-	var queuedAfterScripts = globals.queuedAfterScripts;
-	var superFuncs = globals.superFuncs;
-	var injectsDone = globals.injectsDone;
-
-	globals.kitsy = {
+	// Initialize kitsy
+	bitsy.kitsy = {
 		inject: inject,
 		before: before,
-		after: after
+		after: after,
+		queuedInjectScripts: [],
+		queuedBeforeScripts: [],
+		queuedAfterScripts: []
```

Comment:
ah good catch; it still works because arrays are also just objects, but having them as arrays could lead to potentially confusing behaviour, e.g. you could push to the array and it would create a key of "0"; will fix